homeassistant/components/modbus/base_platform.py

```diff
@@ -225,6 +230,10 @@ def unpack_structure_result(self, registers: list[int]) -> str | None:
         # the conversion only when it's absolutely necessary.
         if isinstance(val_result, int) and self._precision == 0:
             return str(val_result)
+        if isinstance(val_result, str):
+            if val_result == "nan":
+                val_result = STATE_UNAVAILABLE  # pragma: no cover
```

Comment:
`STATE_UNAVAILABLE` should never be set directly as state. We use the entity property `available` set to False to set an entity state as unavailable.

Reply:
@MartinHjelmare 
Something is not right, when doing it I get:
```
  File "/Users/jan/repos/core/homeassistant/components/modbus/base_platform.py", line 193, in __process_raw_value
    self.available = False
    ^^^^^^^^^^^^^^
AttributeError: property 'available' of 'ModbusRegisterSensor' object has no setter
```

Reply:
It's a property, so it can't be set as an attribute. It needs to return a value. You can use `self._attr_available` if you want to set the value directly.

https://developers.home-assistant.io/docs/core/entity#property-implementation